### DISCUSSION TITLE
Foundry Fundamentals: Lesson 17 - Contract not deploying

### DISCUSSION THREAD

**Discussion by FunPeter** [Cyfrin/foundry-full-course-cu#2833] [General]
I am trying to deploy a contract to anvil on foundry in my vs code terminal after running 
```forge script script/DeploySimpleStorage.s.sol``` command but it shows me this 

```[⠒] Compiling...
No files changed, compilation skipped
Error: Could not find target contract
```
. what could possibly be wrong and how do I come up with a solution? By the way I'm using windows OS.

**Comment by 5hr1ganesh** [comment]
The "Could not find target contract" error usually indicates that the script cannot locate the contract it's supposed to deploy. you should

Check Contract Path to Ensure that the path to your contract file is correct. Double-check that DeploySimpleStorage.s.sol is in the `script` directory.

Verify Contract Name to Make sure the contract name in your script matches the actual contract name in the Solidity file.

Clear Previous Builds because Sometimes, old build artifacts can cause issues. Try clearing your build directory with forge clean and then rebuild with forge build.

Check Imports to Ensure that all necessary imports in your contract are correct and that there are no missing files

**Comment by EngrPips** [comment]
Thanks for the heads-up @5hr1ganesh
